torch/csrc/jit/script/init.cpp

```diff
@@ -173,6 +173,24 @@ std::shared_ptr<PythonResolver> pythonResolver(
   return std::make_shared<PythonResolver>(
       rcb, std::move(classname), std::move(classType));
 }
+
+void checkOverloadDecl(const Decl& new_decl, const Decl& old_decl) {
+  const auto& new_params = new_decl.params();
+  const auto& old_params = old_decl.params();
+
+  TORCH_INTERNAL_ASSERT(
```

Comment:
It doesn't really matter for our use case but add a comment here that this isn't strictly necessary, it looks like the overloads can have [different signatures](https://mypy.readthedocs.io/en/latest/more_types.html#function-overloading)

Reply:
Yup, this is along the same path as not requiring default params on the overload that is a follow up